File: app/src/main/java/com/example/purepawapp/ui/admin/AdminBookingsFragment.java
package com.example.purepawapp.ui.admin;

import android.os.Bundle;
import android.view.View;

import androidx.navigation.fragment.NavHostFragment;
import androidx.recyclerview.widget.LinearLayoutManager;

import com.example.purepawapp.R;
import com.example.purepawapp.data.model.Booking;
import com.example.purepawapp.data.repository.RepoCallback;
import com.example.purepawapp.databinding.FragmentAdminBookingsBinding;
import com.example.purepawapp.di.ServiceLocator;
import com.example.purepawapp.ui.common.BaseFragment;
import com.example.purepawapp.util.SpacingItemDecoration;
import com.example.purepawapp.util.ViewUtils;

import java.util.List;

public class AdminBookingsFragment extends BaseFragment<FragmentAdminBookingsBinding> {

    private final AdminBookingAdapter adapter = new AdminBookingAdapter(booking -> {
        Bundle args = new Bundle();
        args.putString("bookingId", booking.getId());
        NavHostFragment.findNavController(this).navigate(R.id.action_adminBookingsFragment_to_adminBookingDetailFragment, args);
    });

    public AdminBookingsFragment() {
        super(FragmentAdminBookingsBinding::inflate);
    }

    @Override
    public void onViewCreated(View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        getBinding().rvBookings.setLayoutManager(new LinearLayoutManager(requireContext()));
        getBinding().rvBookings.setAdapter(adapter);
        getBinding().rvBookings.addItemDecoration(new SpacingItemDecoration((int) (12 * getResources().getDisplayMetrics().density)));
        loadBookings();
    }

    @Override
    public void onResume() {
        super.onResume();
        loadBookings();
    }

    private void loadBookings() {
        ServiceLocator.getSpaRepository().getAllBookings(new RepoCallback<>() {
            @Override
            public void onSuccess(List<Booking> bookings) {
                adapter.submitList(bookings);
                getBinding().emptyState.setVisibility(bookings.isEmpty() ? View.VISIBLE : View.GONE);
            }

            @Override
            public void onError(Exception error) {
                ViewUtils.toast(AdminBookingsFragment.this, error.getMessage() != null ? error.getMessage() : "Không thể tải danh sách lịch hẹn");
            }
        });
    }
}

File: app/src/main/java/com/example/purepawapp/ui/admin/AdminBookingsFragment.kt
package com.example.purepawapp.ui.admin

import android.os.Bundle
import android.view.View
import androidx.core.os.bundleOf
import androidx.lifecycle.lifecycleScope
import androidx.navigation.fragment.findNavController
import androidx.recyclerview.widget.LinearLayoutManager
import com.example.purepawapp.R
import com.example.purepawapp.databinding.FragmentAdminBookingsBinding
import com.example.purepawapp.di.ServiceLocator
import com.example.purepawapp.ui.common.BaseFragment
import com.example.purepawapp.util.SpacingItemDecoration
import com.example.purepawapp.util.toast
import kotlinx.coroutines.launch

class AdminBookingsFragment : BaseFragment<FragmentAdminBookingsBinding>(FragmentAdminBookingsBinding::inflate) {

    private val adapter = AdminBookingAdapter(
        onClick = { booking ->
            findNavController().navigate(
                R.id.action_adminBookingsFragment_to_adminBookingDetailFragment,
                bundleOf("bookingId" to booking.id)
            )
        }
    )

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        binding.rvBookings.layoutManager = LinearLayoutManager(requireContext())
        binding.rvBookings.adapter = adapter
        binding.rvBookings.addItemDecoration(SpacingItemDecoration((12 * resources.displayMetrics.density).toInt()))
        loadBookings()
    }

    override fun onResume() {
        super.onResume()
        loadBookings()
    }

    private fun loadBookings() {
        viewLifecycleOwner.lifecycleScope.launch {
            ServiceLocator.spaRepository.getAllBookings()
                .onSuccess { bookings ->
                    adapter.submitList(bookings)
                    binding.emptyState.visibility = if (bookings.isEmpty()) View.VISIBLE else View.GONE
                }
                .onFailure { toast(it.message ?: "Không thể tải danh sách lịch hẹn") }
        }
    }
}
